frontend/modules/recipe_form/components/RecipeForm.js

```diff
@@ -50,7 +53,15 @@ class RecipeForm extends React.Component {
   }
 
   _onInit() {
-    this.setState( this.getStateFromStore() );
+    const state = this.getStateFromStore();
+    this.setState(state);
+
+    if (Object.keys(state.data).length > 0) {
+      const user = this.getAuthUser()
+      if (state.data.author !== user.id) {
+        browserHistory.push('/');
```

Comment:
A few things here:

1. Line 60 needs a semi-colon.
2. Line 61 is having the same user check issue. `user` is null.
3. Doing a push here breaks the browse history. If a user tries to go to the edit page, then hits the back button, its a never ending loop.

IMHO, when a user tries to go to the edit page without permission it should redirect them to the recipe view page. While also keeping the browser history intact.